### PR TITLE
ENG-1851 Use elkjs for positioning nodes and edges

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR adopts the elkjs library to handle node and edge positioning, allowing us a bit more flexibility and customization in how we should render our DAGs.

## Related issue number (if any)
ENG-1851

## Loom demo (if any)
https://www.loom.com/share/9d8630c722754346a0db01fcc3fce9f2

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/workflows/StatusBar.tsx [hunk 1]
```diff
@@ -278,18 +278,13 @@ export const WorkflowStatusBar: React.FC<WorkflowStatusBarProps> = ({
...
-  // };
-
-  const itemsToShow: WorkflowStatusItem[] = workflowStatusItems.filter(
-    (statusItem: WorkflowStatusItem) => {
-      return statusItem.type !== 'paramOp';
-    }
-  );
-
   useEffect(() => {
+    const itemsToShow = workflowStatusItems.filter(
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/StatusBar.tsx:282]: This file's changes are unrelated to the ENG-1851 task, but take care of a linter warning.

File: src/ui/common/src/components/workflows/nodes/BaseNode.styles.tsx [hunk 2]
```diff
@@ -10,9 +10,10 @@ const BaseNode = styled(Box)({
   borderStyle: 'solid',
   borderWidth: '2px',
   padding: '10px',
-  maxWidth: '250px',
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/nodes/BaseNode.styles.tsx:13]: Going to revert these sizes back

File: src/ui/common/src/components/pages/metric/id/index.tsx [hunk 3]
```diff
@@ -59,8 +59,6 @@ const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
       state.workflowDagResultsReducer.results[workflowDagResultId]
   );
 
-  const workflow = useSelector((state: RootState) => state.workflowReducer);
```
Comment (agiron123) [inline src/ui/common/src/components/pages/metric/id/index.tsx:62]: takes care of a linter warning.

File: src/ui/common/src/components/workflows/nodes/Node.tsx [hunk 4]
```diff
@@ -19,6 +19,9 @@ type Props = {
   icon?: IconDefinition;
 };
 
+export const NODE_WIDTH = 250;
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/nodes/Node.tsx:22]: Remove these, they're not being used any more.

File: src/ui/common/src/reducers/workflow.ts [hunk 5]
```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
...
+          'org.eclipse.elk.edgeRouting': 'SPLINES',
+          //https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-nodePlacement-strategy.html
+          'crossingMinimization.strategy': 'INTERACTIVE',
+          'crossingMinimization.forceNodeModelOrder': true,
+        },
+        children: mappedNodes,
+        edges: mappedEdges,
+      };
+
+      console.log('mappedEdges: ', mappedEdges);
```
Comment (agiron123) [inline src/ui/common/src/reducers/workflow.ts:504]: remove log

File: src/ui/common/src/reducers/workflow.ts [hunk 6]
```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
...
+              };
+
+              // TODO (future work): Handle mapping elk edge sections to react-flow edges
+            }
+          }
+
+          return node;
+        });
+      } catch (error) {
+        console.log('error getting PositionedLayout: ', error);
```
Comment (agiron123) [inline src/ui/common/src/reducers/workflow.ts:523]: TODO: change this to show an error alert instead so the user knows that something failed.

File: src/ui/common/src/reducers/workflow.ts [hunk 7]
```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
...
+        const positionedLayout = await elk.layout(graph);
+        collapsedPosition.nodes = collapsedPosition.nodes.map((node) => {
+          for (let i = 0; i < positionedLayout.children.length; i++) {
+            if (node.id === positionedLayout.children[i].id) {
+              node.position = {
+                x: positionedLayout.children[i].x,
+                y: positionedLayout.children[i].y,
+              };
+
+              // TODO (future work): Handle mapping elk edge sections to react-flow edges
```
Comment (saurav-c) [inline src/ui/common/src/reducers/workflow.ts:516]: nit: add a linear task

File: src/ui/common/src/reducers/workflow.ts [hunk 8]
```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
...
+          id: node.id,
+          // Width and height set not only the node width/height, but also how far apart we want to position these nodes.
+          // So using a value of just the width and the height of the node will result in no spacing in between.
+          // Elk (eclipse layout kernel) also provides many knobs to tweak this as well.
+          width: 300,
+          height: 300,
+        };
+      });
+
+      // TODO: move into collapsed nodes function.
```
Comment (saurav-c) [inline src/ui/common/src/reducers/workflow.ts:471]: nit: add a linear task

**Review by saurav-c — APPROVED** [review]
LGTM

File: src/ui/common/src/reducers/workflow.ts (hunk 8 above)
Comment (agiron123) [reply src/ui/common/src/reducers/workflow.ts:471]: Done!
https://linear.app/aqueducthq/issue/ENG-2303/chore-move-logic-for-filtering-out-missing-edges-into-collapse-nodes

File: src/ui/common/src/reducers/workflow.ts (hunk 7 above)
Comment (agiron123) [reply src/ui/common/src/reducers/workflow.ts]: done 
https://linear.app/aqueducthq/issue/ENG-2305/handle-mapping-elk-edge-sections-to-react-flow-edges

File: src/ui/common/src/reducers/workflow.ts (hunk 6 above)
Comment (agiron123) [reply src/ui/common/src/reducers/workflow.ts:522]: ENG-2304

**Pull request merged**